src/reconstruction/community.jl

```diff
@@ -63,35 +63,38 @@ end
     add_model_with_exchanges(
         community::CoreModel,
         model::MetabolicModel,
-        exchange_rxn_ids::Vector{String},
-        exchange_met_ids::Vector{String};
+        exchange_rxn_mets::Dict{String,String};
         model_name = "unknown_species",
         biomass_id = nothing,
     )::CoreModel
 
-Add `model` to `community`, which is a pre-existing community model with
-`exchange_rxn_ids` and `exchange_met_ids`. The `model_name` is appended to
-each reaction and metabolite, see [`join_with_exchanges`](@ref). If
-`biomass_id` is specified then a biomass metabolite for `model` is also added
-to the resulting model. The column corresponding to the `biomass_id` reaction
-then produces this new biomass metabolite with unit coefficient. Note,
-`exchange_rxn_ids` and `exchange_met_ids` must already exist in the `community`
-model.
+Add `model` to `community`, which is a pre-existing community model with exchange reactions
+and metabolites in the dictionary `exchange_rxn_mets`. The `model_name` is appended to each
+reaction and metabolite, see [`join_with_exchanges`](@ref). If `biomass_id` is specified
+then a biomass metabolite for `model` is also added to the resulting model. The column
+corresponding to the `biomass_id` reaction then produces this new biomass metabolite with
+unit coefficient. Note, `exchange_rxn_ids` and `exchange_met_ids` must already exist in the
+`community` model.
 
 # Example
 ```
-community = add_model_with_exchanges(community, model, exchange_rxn_ids, exchange_met_ids; model_name="species_2", biomass_id="BIOMASS_Ecoli_core_w_GAM")
+community = add_model_with_exchanges(community,
+    model,
+    exchange_rxn_mets;
+    model_name="species_2",
+    biomass_id="BIOMASS_Ecoli_core_w_GAM")
 ```
 """
 function add_model_with_exchanges(
     community::CoreModel,
     model::MetabolicModel,
-    exchange_rxn_ids::Vector{String},
-    exchange_met_ids::Vector{String};
+    exchange_rxn_mets::Dict{String,String};
     model_name = "unknown_species",
     biomass_id = nothing,
 )::CoreModel
 
+    exchange_rxn_ids = keys(exchange_rxn_mets)
+    exchange_met_ids = [exchange_rxn_mets[k] for k in exchange_rxn_ids]
```

Comment:
I found in documentation of `?values` that keys() and values() are actually guaranteed to return the elements in the same order, so this may get simplified.

It sucks that there's no obvious way to "unzip" the dictionary.

Reply:
lol, will change :)